streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobGenerator.scala

```diff
@@ -221,8 +221,12 @@ class JobGenerator(jobScheduler: JobScheduler) extends Logging {
     logInfo("Batches pending processing (" + pendingTimes.size + " batches): " +
       pendingTimes.mkString(", "))
     // Reschedule jobs for these times
-    val timesToReschedule = (pendingTimes ++ downTimes).filter { _ < restartTime }
-      .distinct.sorted(Time.ordering)
+    val skipDownTime = conf.getBoolean("spark.streaming.skipDownTimeBatch", false)
```

Comment:
I'd prefer not to add yet another configuration to control this. It adds complexity. I don't think the name is descriptive here; what is a 'down time batch'? The current behavior is coherent, since the expected behavior is to pick up where it left off. It's not intended that you leave the job not running for a long time relative to the batch interval.